core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -484,10 +484,7 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
    * group does not know, because the information is not available yet or because the it has
    * failed to parse the Consumer Protocol, it returns true to be safe.
    */
-  def isSubscribedToTopic(topic: String): Boolean = subscribedTopics match {
-    case Some(topics) => topics.contains(topic)
-    case None => true
-  }
+  def isSubscribedToTopic(topic: String): Boolean = subscribedTopics.forall(_.contains(topic))
```

Comment:
FYI reviewers:

This method is only called by `handleOffDeleteRequest` which is not a performance sensitive path and hence, unnecessary object allocation in absence of `inline-scala` compilation flag should be ok. Opting to favour readibility and conciseness of code here.